### PR TITLE
test(e2e): retain traces for failed runs

### PR DESCRIPTION
**Pull request by raducristianpopa**
## Context

With the current setup, if the first run fails we are not saving the trace for it, only for the first retry.
## Changes proposed in this pull request

Retain traces only for failed runs.

### CODE REVIEW

**Review by raducristianpopa — COMMENTED** [review]
test-e2e

File: playwright.config.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ export default defineConfig({
       { open: 'never', outputFolder: path.join(testDir, 'playwright-report') },
     ],
   ],
-  use: { trace: 'on-first-retry' },
+  use: { trace: 'retain-on-failure' },
```
Comment (sidvishnoi) [inline playwright.config.ts:29]: Or do we want this?
```suggestion
  use: { trace: 'on' },
```

Comment (sidvishnoi) [reply playwright.config.ts:29]: No, checked docs. One you added is best one.

Comment (raducristianpopa) [reply playwright.config.ts:29]: I don't mind having this. Are we going to use the traces from successful runs as well?

Comment (sidvishnoi) [reply playwright.config.ts:29]: No. I just checked docs. retain-on-failure is best.

**Review by raducristianpopa — COMMENTED** [review]
test-e2e

**Review by sidvishnoi — APPROVED** [review]

**Pull request merged**